core/src/main/java/hudson/DNSMultiCast.java

```diff
@@ -97,6 +97,21 @@ public void close() {
 
     public static boolean disabled = SystemProperties.getBoolean(DNSMultiCast.class.getName()+".disabled");
 
+    /**
+     * Class that extends {@link JmDNSImpl} to add an abort method. Since {@link javax.jmdns.JmDNS#close()} might
+     * make the instance hang during the shutdown, the abort method terminate uncleanly, but rapidly and
+     * without blocking.
+     *
+     * Initially it was part of the jenkinsci/jmdns forked library, but now this class is responsible for aborting,
+     * allowing to have a direct and clean dependency to the original library.
+     *
+     * The abort() method is pretty similar to close() method. To access private methods and fields uses
+     * reflection.
+     *
+     * @since 2.178
```

Comment:
Not needed for private classes

Reply:
I know the `@since` annotation is not needed for private classes, but in this case I've added because we are deprecating the jenkinsci fork of jmdns and I think it could be interesting to keep the trace of when that happened just with a glance. I would be happy to remove it tough